File: ggalmazor/php/src/Lonja.php
<?php

class Lonja {

  private $ciudad;
  private $distancia;
  private $precios;

  public function __construct($ciudad, $distancia, $precios) {
    $this->ciudad = $ciudad;
    $this->distancia = $distancia;
    $this->precios = $precios;
  }

  public function porCuantoSeVendeElLoteDe($lote) {
    $precioLote = 0;
    foreach ($lote as $producto => $kilogramos) {
      $precioLote = $this->aCuantoEstaElKiloDe($producto) * $kilogramos;
    }
    return $precioLote;
  }
  
  public function aQueDistanciaEsta() {
    return $this->distancia;
  }
  
  public function ciudad() {
    return $this->ciudad;
  }

  private function aCuantoEstaElKiloDe($producto) {
    return $this->precios[$producto];
  }

}

File: jasosa/KataLonja/Lonja.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace KataLonja
{
    public class Lonja
    {
        /// <summary>
        /// Ciudad de la lonja
        /// </summary>
        public NombreCiudad Ciudad { get; set; }

        /// <summary>
        /// Distancia en Km desde Santiago
        /// </summary>
        public float Distancia { get; set; }

        /// <summary>
        /// Precios de compra de los diferentes productos en la lonja
        /// </summary>
        public List<ValorDeCompra> Precios { get; set; }

        internal float CalculaPrecioVenta(Carga carga, Depreciación depreciacion)
        {
            float precioVentaTotal = 0;
            foreach (ValorDeCompra precioPorProducto in Precios)
            {
                precioVentaTotal += carga.Peso(precioPorProducto.Producto) * depreciacion.Calcula(Distancia, precioPorProducto.Importe);                
            }

            return precioVentaTotal;
        }
    }
}
